File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/entity/ReferralDetails.kt
package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity

import java.time.LocalDate
import java.time.OffsetDateTime
import java.util.UUID
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class ReferralDetails(
  @Id val id: UUID,
  var supersededById: UUID?,
  val referralId: UUID,
  val createdAt: OffsetDateTime,
  @Column(name = "created_by") val createdByUserId: String,
  val reasonForChange: String?,

  // actual referral details fields:
  var completionDeadline: LocalDate?,
  var furtherInformation: String?,
  var maximumEnforceableDays: Int?,
)

File: server/models/referralDetails.ts
interface ReferralDetailsFields {
  maximumEnforceableDays?: number | null
  completionDeadline?: string | null
  furtherInformation?: string | null
  reasonForReferral?: string | null
  reasonForReferralFurtherInformation?: string | null
}

export default interface ReferralDetails extends ReferralDetailsFields {
  referralId: string
}

export interface ReferralDetailsUpdate extends ReferralDetailsFields {
  reasonForChange: string
  expectedProbationOffice?: string | null
}
